generator/main.py

```diff
@@ -491,26 +460,106 @@ def main():
+    parser.add_argument(kExtraPythonPath, default=None, type=str, help='Additional python path to set before loading Jinja2')
+    parser.add_argument('--output-json-tarball', default=None, type=str, help='Name of the "JSON tarball" to create (tar is too annoying to use in python).')
+    parser.add_argument('--depfile', default=None, type=str, help='Name of the Ninja depfile to create for the JSON tarball')
+    parser.add_argument('--expected-outputs-file', default=None, type=str, help="File to compare outputs with and fail if it doesn't match")
+    # Arguments used only for the CMake build
+    parser.add_argument('-o', '--output-dir', default=None, type=str, help='Output directory for the generated source files.')
+    parser.add_argument('--print-dependencies', action='store_true', help='Prints a space separated list of file dependencies, used for CMake integration')
+    parser.add_argument('--print-outputs', action='store_true', help='Prints a space separated list of file outputs, used for CMake integration')
+
+    args = parser.parse_args()
+
+    # Load and parse the API json file
+    with open(args.json[0]) as f:
+        loaded_json = json.loads(f.read())
+    api_params = parse_json(loaded_json)
+
+    targets = args.targets.split(',')
+    renders = get_renders_for_targets(api_params, targets)
+
+    # Print outputs and dependencies for CMake
     if args.print_dependencies:
         dependencies = set(
             [os.path.abspath(args.template_dir + os.path.sep + render.template) for render in renders] +
             [os.path.abspath(args.json[0])] +
             [os.path.realpath(__file__)]
         )
         dependencies = [dependency.replace('\\', '/') for dependency in dependencies]
-        sys.stdout.write(output_separator.join(dependencies))
+        sys.stdout.write(';'.join(dependencies))
         return 0
 
     if args.print_outputs:
         outputs = set(
             [os.path.abspath(args.output_dir + os.path.sep + render.output) for render in renders]
         )
         outputs = [output.replace('\\', '/') for output in outputs]
-        sys.stdout.write(output_separator.join(outputs))
+        sys.stdout.write(';'.join(outputs))
         return 0
 
-    do_renders(renders, args.template_dir, args.output_dir)
+    # The caller wants to assert that the outputs are what it expect.
+    # Load the file and compare with our renders. GN-only.
+    if args.expected_outputs_file != None:
+        with open(args.expected_outputs_file) as f:
+            expected = set([line.strip() for line in f.readlines()])
+
+        actual = set()
+        actual.update([render.output for render in renders])
+
+        if actual != expected:
+            print("Wrong expected outputs, caller expected:\n    " + repr(list(expected)))
+            print("Actually output:\n    " + repr(list(actual)))
+            return 1
+
+    outputs = do_renders(renders, args.template_dir)
+
+    # CMake only, output all the files directly.
```

Comment:
nit: suggest : instead of , (possible comma splice)

Reply:
done